src/components/TreeViewFinder/TreeViewFinder.jsx

```diff
@@ -187,6 +193,45 @@ const TreeViewFinder = (props) => {
         // will proc onNodeSelect then ...
     };
 
+    useEffect(() => {
+        if (selectedProp?.length > 0) {
+            setSelected((oldSelectedNodes) => [
+                ...oldSelectedNodes,
+                ...selectedProp,
+            ]);
+        }
+    }, [selectedProp]);
+
+    useEffect(() => {
+        if (expandedProp?.length > 0) {
+            setExpanded((oldExpandedNodes) => [
+                ...oldExpandedNodes,
+                ...expandedProp,
+            ]);
+        }
+    }, [expandedProp]);
+
+    useEffect(() => {
+        // if we have selected elements by default, we scroll to it
+        if (selectedProp?.length > 0) {
+            // we check if all expanded nodes by default all already expanded first
+            const isNodeExpanded = expandedProp.every((nodeId) =>
+                expanded.includes(nodeId)
+            );
+
+            // we got the last element that we suppose to scroll to
+            const lastScrollRef =
+                scrollRef.current[scrollRef.current.length - 1];
+            if (isNodeExpanded && lastScrollRef) {
+                lastScrollRef.scrollIntoView({
+                    behavior: 'smooth',
+                    block: 'center',
+                    inline: 'center',
+                });
+            }
+        }
+    }, [expanded, selectedProp, expandedProp, data]);
```

Comment:
seems dangerous...
each time expanded will change, this effect will proc, and we could have strange scroll behaviours maybe...